File: AT-2.java/src/Hija/Automovil.java
/*
 Crear una clase hija llamada Automóvil que herede de la clase Vehículo y que tenga 
   los atributos marca, modelo y precio, crear sus respectivas funciones para obtener 
   los datos y mostrar datos. 
 */
package Hija;
import Padre.Vehiculo;

/**
 *
 * @author LENOVO
 */
public class Automovil extends Vehiculo {
    
    String marca;
    String modelo;
    int precio;

    public Automovil(String fecha_Fbn, String vin_Chas, String vin_Motor,
            String marca, String modelo, int precio) {
        
        super(fecha_Fbn, vin_Chas, vin_Motor);
        this.marca = marca;
        this.modelo = modelo;
        this.precio = precio;
    }

    public String getMarca() {
        return marca;
    }

    public String getModelo() {
        return modelo;
    }

    public int getPrecio() {
        return precio;
    }
    
    public void mostrarDatos() {
  
        System.out.println("Su Fecha de fabricacion: " + getFechaFabricacion());
        System.out.println("Su VIN Chasis: " + getVinChasis());
        System.out.println("Su VIN Motor: " + getVinMotor());
        System.out.println("Su Marca: " + getMarca());
        System.out.println("Su Modelo: " + getModelo());
        System.out.println("Su Precio: $" + getPrecio());
    }
}

File: AT-2.py/Automovil.py
from Vehiculo import Vehiculo 

class Automovil(Vehiculo):
    def __init__(self, fecha_fabricacion, vin_chasis, vin_motor,
                 marca, modelo, precio):
        super().__init__(fecha_fabricacion, vin_chasis, vin_motor)
        self.marca = marca
        self.modelo = modelo
        self.precio = precio

    def get_marca(self):
        return self.marca

    def get_modelo(self):
        return self.modelo

    def get_precio(self):
        return self.precio

    def mostrar_datos(self):
        print(f"Su Fecha de fabricación: {self.get_fecha_fabricacion()}")
        print(f"Su VIN Chasis: {self.get_vin_chasis()}")
        print(f"Su VIN Motor: {self.get_vin_motor()}")
        print(f"Su Marca: {self.get_marca()}")
        print(f"Su Modelo: {self.get_modelo()}")
        print(f"Su Precio: ${self.get_precio()}")
